Go1.13 golang.org/cl/161199 changes text/scanner error message affecting lexer

Heads up, https://golang.org/cl/161199 updates text/scanner error messages and breaks the assumption in the following code --

https://github.com/alecthomas/participle/blob/master/lexer/text_scanner.go#L56

```sh
$ go test ./...
ok      github.com/alecthomas/participle        0.067s
--- FAIL: TestLexSingleString (0.00s)
    require.go:794:
                Error Trace:    text_scanner_test.go:35
                Error:          Received unexpected error:
                                <source>:1:14: invalid char literal
                Test:           TestLexSingleString
FAIL
FAIL    github.com/alecthomas/participle/lexer  0.033s
ok      github.com/alecthomas/participle/lexer/ebnf     0.045s
ok      github.com/alecthomas/participle/lexer/ebnf/internal    0.019s
```

I can send a PR to update the error message comparison or someone else can come up with a better way of not relying on the error message string.

--- a/lexer/text_scanner.go
+++ b/lexer/text_scanner.go
@@ -53,7 +53,7 @@ func Lex(r io.Reader) Lexer {
 	lexer := lexWithScanner(r, &scanner.Scanner{})
 	lexer.scanner.Error = func(s *scanner.Scanner, msg string) {
 		// This is to support single quoted strings. Hacky.
-		if msg != "illegal char literal" {
+		if !strings.HasSuffix(msg, "char literal") {
 			lexer.err = Errorf(Position(lexer.scanner.Pos()), msg)
 		}
 	}

